pandas/core/generic.py

```diff
@@ -3179,6 +3179,481 @@ def to_csv(
     # ----------------------------------------------------------------------
     # Fancy Indexing
 
+    @property
```

Comment:
so to avoid moving these multiple times, can you create a mix-in class where these are put (you can define this in pandas.core.indexing i think would be ok), then just include it in the classes for Series/DataFrame

Reply:
maybe better (as we are going to do this anyhow) is create

pandas/core/generic/indexing.py (and put the mxin there)

Reply:
Good idea with a Mixin. Why do you want to set it on DataFrame/Series directly? Seems simpler to just set it on NDFrame.

Reply:
right that’s totally fine